Remove Matter from sdk-nrf (#30253)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/software_maturity.rst

Reviewer: ```suggestion
           - --\ :sup:`8`
```

Reviewer 2: ```suggestion
- --\ :sup:`8`
- --\ :sup:`8`
- --\ :sup:`8`
- --\ :sup:`8`
```

           - --
           - --

>            - --

Reviewer 2: ```suggestion
- --\ :sup:`8`
- --\ :sup:`8`
- --\ :sup:`8`
- --\ :sup:`8`
- --\ :sup:`8`
```

Matter features support
***********************

> The software maturity levels of the support for each Matter feature can be found on the `Matter Add-on`_ documnetation.

Reviewer: ```suggestion
The software maturity levels of the support for each Matter feature can be found in the `Matter Add-on`_ documentation.
```
